Donation-platform/src/main/java/com/google/FindTrendingCharities.java

```diff
@@ -30,144 +30,155 @@
-    } catch (EntityNotFoundException e) {
-      System.out.println("tag entity not found: " + e);
-    } catch (TooManyResultsException e) {
-      System.out.println("duplicate tags exist in db: " + e);
-    } catch (Exception e) {
-      System.out.println("unexpected exception: e");
+
+    private boolean hasCharityNavRating(Charity charity) {
+        return false;
     }
-    double charityTrendingScore =
-        TAGS_SCORE_WEIGHT * charityTagsScore + AVG_REVIEW_WEIGHT * avgReview;
-    return charityTrendingScore;
-  }
-
-  // TODO: Decide whether we will be using charityNavRating and change accordingly
-  private double calcCharityNavRating(Charity charity) {
-    // double charityNavRating;
-    // return charityNavRating * CHARITY_NAV_SCALE_FACTOR;
-    return 0;
-  }
-
-  private boolean hasCharityNavRating(Charity charity) {
-    return false;
-  }
-
-  // return the average trending score of a collection of tags
-  private double getTagTrendingScore(Collection<Tag> tags) throws Exception {
-    double sumScores = 0;
-    int numTags = tags.size();
-    for (Tag tag : tags) {
-      double tagScore = tag.getTrendingScoreTag();
-
-      // TODO: Use GoogleTrends API to update tag trending score
-
-      sumScores += tagScore;
+
+    // return the average trending score of a collection of tags
+    private double getTagTrendingScore(Collection<Tag> tags) throws Exception {
+        double sumScores = 0;
+        int numTags = tags.size();
+        for (Tag tag : tags) {
+        double tagScore = tag.getTrendingScoreTag();
+        sumScores += tagScore;
+        }
+        return (sumScores / numTags);
     }
-    return (sumScores / numTags);
-  }
 
+    //return Collection of all Tags in db
     public Collection<Tag> getTagsDb() throws Exception{
         Collection<Tag> tags= db.getAllTags();
         return tags;
     }
+    
+    //updates one tag in db with a new trendingScore
+    public void updateTagScore(Tag tagToUpdate) throws Exception {
+        db.updateTag(tagToUpdate);
+    }
+
+    //update all charity trending scores
+    public List<Charity> updateCharityScores() {
+        for (Charity charity : charities) {
+            double charityScore = calcCharityTrendingScore(charity);
+            charity.setTrendingScoreCharity(charityScore);
+            try {
+                db.updateCharity(charity);
+            }
+            catch (Exception e) {
+                System.out.println("unable to update charity: " + e);
+            }
+        }
+        System.out.println("updated charities");
+        //List<Charity> trendingCharities = sortTrendingCharities();
+        sortTrendingCharities();
+        return topTrendingCharities;
+    }
 
-    public void updateTagScores(Collection<Tag> tagsToUpdate) throws Exception {
-        for (Tag tag: tagsToUpdate) {
-            db.updateTag(tag);
+    //keep topTrendingCharities ready for request to return
+    public void sortTrendingCharities() {
```

Comment:
this function should be generic and just sortCharities. It should take list of charities as an argument, sort it and return. move the logic to find charities outside.